contracts/provider/assembly/index.ts

```diff
@@ -0,0 +1,216 @@
+import {
+  u128,
+  storage,
+  context,
+  ContractPromiseBatch,
+  PersistentUnorderedMap,
+  logging,
+} from "near-sdk-as";
+import { onlyOwner } from "../../ownable/assembly";
```

Comment:
Note that we _use_ onlyOwner here, and then we also export the relevant public methods from ownable so that this contract essentially _becomes_ ownable!